lib/matplotlib/backends/backend_gtk3cairo.py

```diff
@@ -36,16 +41,11 @@ def _render_figure(self, width, height):
     def on_draw_event(self, widget, ctx):
         """ GtkDrawable draw event, like expose_event in GTK 2.X
         """
-        toolbar = self.toolbar
-        if toolbar:
-            toolbar.set_cursor(cursors.WAIT)
-        self._renderer.set_context(ctx)
-        allocation = self.get_allocation()
-        x, y, w, h = allocation.x, allocation.y, allocation.width, allocation.height
-        self._render_figure(w, h)
-        if toolbar:
-            toolbar.set_cursor(toolbar._lastCursor)
-        return False  # finish event propagation?
+        with (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
+              else ExitStack()):
```

Comment:
`self.toolbar._wait_cursor_for_draw_cm() if self.toolbar else ExitStack()` happens here the second time, `ExitStack` is overkill for this usage, I think it would be nice to move this to a function and do not bring a `contextlib2` dependency.

Reply:
It's not a second time; this is gtk3*cairo* which goes through a different rendering path than the various agg backends.

Using ExitStack as a "do-nothing" context manager to optionally enter a context is explicitly mentioned in the docs: https://docs.python.org/3/library/contextlib.html#simplifying-support-for-single-optional-context-managers.  contextlib2 is a new dependency only on Py2; I think this is perfectly idiomatic Py3 (and again Py2 is very low on my priority list...).  It's also a pure python dependency so it's not as if installing it was particularly difficult.

Reply:
In Python 3.7, there will be an explicit `nullcontext()` for that purpose, which is semantically more clear.
https://docs.python.org/3/library/contextlib.html#simplifying-support-for-single-optional-context-managers

Please add a comment concerning this so that we can replace it once we'll depend on 3.7+ only.

Alternatively:
~~~
try:
    from contextlib import nullcontext
except ImportError:  # python 3.6 backward compatibility
    from contextlib import ExitStack as nullcontext
~~~